src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -0,0 +1,541 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AppCommandlineArgs.h"
+#include "ActionArgs.h"
+
+using namespace winrt::TerminalApp;
+using namespace TerminalApp;
+
+// Either a ; at the start of a line, or a ; preceeded by any non-\ char.
+const std::wregex AppCommandlineArgs::_commandDelimiterRegex{ LR"(^;|[^\\];)" };
+
+AppCommandlineArgs::AppCommandlineArgs()
+{
+    _buildParser();
+    _resetStateToDefault();
+}
+
+// Method Description:
+// - Attempt to parse a given command as a single commandline. If the command
+//   doesn't have a subcommand, we'll try parsing the commandline again, as a
+//   new-tab command.
+// - Actions generated by this command are added to our _startupActions list.
+// Arguments:
+// - command: The individual commandline to parse as a command.
+// Return Value:
+// - 0 if the commandline was successfully parsed
+int AppCommandlineArgs::ParseCommand(const Commandline& command)
+{
+    const int argc = static_cast<int>(command.Argc());
+
+    try
+    {
+        // CLI11 needs a mutable vector<string>, so copy out the args here.
+        // * When we're using the vector<string> parse(), it also expects that
+        //   there isn't a leading executable name in the args, so slice that
+        //   out.
+        //   - In AppCommandlineArgs::BuildCommands, we'll make sure each
+        //     subsequent command in a single commandline starts with a wt.exe.
+        //     Our very first argument might not be "wt.exe", it could be `wt`,
+        //     or `wtd.exe`, etc. Regardless, we want to ignore the first arg of
+        //     every Commandline
+        // * Not only that, but this particular overload of parse() wants the
+        //   args _reversed_ here.
+        std::vector<std::string> args{ command.Args().begin() + 1, command.Args().end() };
+        std::reverse(args.begin(), args.end());
+
+        // Revert our state to the initial state. As this function can be called
+        // multiple times during the parsing of a single commandline (once for each
+        // sub-command), we don't want the leftover state from previous calls to
+        // pollute this run's state.
+        _resetStateToDefault();
+
+        // Manually check for the "/?" or "-?" flags, to manually trigger the help text.
+        if (argc == 2 && (NixHelpFlag == command.Args().at(1) || WindowsHelpFlag == command.Args().at(1)))

```

Comment:
the back ones can use til::at(command.Args(), 1) to skip the check since you literally just checked the length of argc.